packages/next-intl/src/navigation/createLocalizedPathnamesNavigation.tsx

```diff
@@ -15,6 +15,17 @@ import {
   HrefOrUrlObjectWithParams
 } from './utils';
 
+export type LocalizedPathnamesLinkProps<
+  Locales extends AllLocales,
+  Pathname extends keyof Pathnames<Locales>
+> = Omit<
+  ComponentProps<typeof BaseLink>,
+  'href' | 'name'
+> & {
+  href: HrefOrUrlObjectWithParams<Pathname>;
+  locale?: Locales[number];
+};
```

Comment:
Hey, thank you so much for the proposal!

From your code snippet in the PR comment:

```tsx
type LinkProps = LocalizedPathnamesLinkProps<keyof typeof pathnames>
```

I noticed as well that the type of the pathnames need to be passed to be able to compose the `href` prop in a wrapping component.

This is briefly mentioned in [the docs](https://next-intl-docs.vercel.app/docs/routing/navigation#link):

<img width="767" alt="Screenshot 2023-09-01 at 08 47 38" src="https://github.com/amannn/next-intl/assets/4038316/f0099a77-4d6a-40d2-a63e-4361ae976c31">

From my perspective the `LocalizedPathnamesLinkProps` helper doesn't do much more than `ComponentProps<typeof Link<Pathname>>`.

I tend to be on the side of not exposing more API than necessary if something can already be achieved somewhat reasonably.

Would that work for you? Can something be improved in regard to the docs?

On a side note, it would of course be fantastic if the `href` can be composed without passing the type argument again, but I couldn't find a way so far.